### ISSUE TITLE
Unreachable code branch

### ISSUE THREAD

**Issue by Nashenas88** [cristicbz/rust-doom#88]
At line 205 [here](https://github.com/cristicbz/rust-doom/blob/master/src/game/player.rs#L205) you can see that the `else if` branch will never be run because it's checking the same condition as the first `if` branch.

Here's the snippet
```rust
            if contact.time < 1.0 {
                (self.height * contact.time, Some(contact.normal))
            } else if contact.time < 1.0 {
                (self.height, Some(contact.normal))
            } else {
                (self.height, None)
            }
```

**Comment by cristicbz** [comment]
Lol, d'oh! Thanks! Now, I wonder what I meant to write there, hmmm.

**Comment by cristicbz** [comment]
Ahhh, I used to have a bit of tolerance, like 1.5 or whatever, and then I didn't, but left the if branch in there. That first branch should be `<=` and the second should just go away.

**Comment by cristicbz** [comment]
Fixed by #90
